py4DSTEM/process/utils/utils.py

```diff
@@ -607,3 +608,204 @@ def get_ewpc_filter_function(Q_Nx, Q_Ny):
     h = np.hanning(Q_Nx)[:,np.newaxis] * np.hanning(Q_Ny)[np.newaxis,:]
     return lambda x: np.abs(np.fft.fftshift(np.fft.fft2(h*np.log(np.maximum(x,0.01)))))**2
 
+
+
+def fourier_resample(
+    array, 
+    scale=None, 
+    output_size=None,
+    force_nonnegative=True,
+    bandlimit_nyquist=None,
+    bandlimit_power=2, 
+    dtype=np.float32):
+    """
+    Resize a 2D array along any dimension, using Fourier interpolation / extrapolation.
+    For 4D input arrays, only the final two axes can be resized.
+
+    The scaling of the array can be specified by passing either `scale`, which sets
+        the scaling factor along both axes to be scaled; or by passing `output_size`, 
+        which specifies the final dimensions of the scaled axes (and allows for different
+        scaling along the x,y or kx,ky axes.)
+
+    Args:
+        array (2D/4D numpy array): Input array, or 4D stack of arrays, to be resized. 
+        scale (float): scalar value giving the scaling factor for all dimensions
+        output_size (2-tuple of ints): two values giving either the (x,y) output size for 2D, or (kx,ky) for 4D
+        force_nonnegative (bool): Force all outputs to be nonnegative, after filtering
+        bandlimit_nyquist (float): Gaussian filter information limit in Nyquist units (0.5 max in both directions)
+        bandlimit_power (float): Gaussian filter power law scaling (higher is sharper)
+        dtype (numpy dtype): datatype for binned array. default is single precision float
+
+    Returns:
+        the resized array (2D/4D numpy array)
+    """
+    input__size = array.shape[-2:]
+
+
+    if scale is not None:
+        assert output_size is None, 'Cannot specify both a scaling factor and output size'
+        assert np.size(scale) == 1, 'scale should be a single value'
+        scale = np.asarray(scale)
+        output_size = (input__size * scale).astype('intp')
+    else:
+        assert scale is None, 'Cannot specify both a scaling factor and output size'
+        assert np.size(output_size) == 2, 'output_size must contain two values'
+        output_size = np.asarray(output_size)
+    
+    scale_output = np.prod(output_size) / np.prod(input__size)
+
+
+    if bandlimit_nyquist is not None:
+        if len(input__size) == 2:
+            kx = np.fft.fftfreq(output_size[0])
+            ky = np.fft.fftfreq(output_size[1])
+        if len(input__size) == 4:
+            kx = np.fft.fftfreq(output_size[2])
+            ky = np.fft.fftfreq(output_size[3])
```

Comment:
I'm pretty sure this if statement isn't doing anything....the way the logic above is set up, both `input__size` and `output__size` will always have 2 elements.

Reply:
good catch, this should be `if np.size(array.shape) == 4:`

Reply:
I think you still don't actually need the if statements checking the length at all.  The way you've written this, output_size should always have length 2, and so regardless of whether we're dealing with 2 or 4D input arrays, you should just be able to use
```
kx = np.fft.fftfreq(output_size[0])
ky = np.fft.fftfreq(output_size[1])
```

Reply:
aha even better, i ingeniously placed the buggy code in an inaccessible branch

Reply:
Correctomundo!